staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -124,6 +127,137 @@ type ConfigFlags struct {
 	discoveryQPS float32
 }
 
+func parseName(str string) string {
+	if len(str) < 1 {
+		// just return an empty string if it's empty
+		return ""
+	}
+	if str[:2] == "--" {
+		return str[2:strings.Index(str, "=")]
+	} else if str[:1] == "-" {
+		return str[1:strings.Index(str, " ")]
+	} else {
+		return ""
+	}
+}
+
+func parseValEqualPtr(str string) *string {
+	val := str[strings.Index(str, "=")+1:]
+	val = val[1 : len(val)-1]
+	return &val
+}
+
+func parseValEqual(str string) string {
+	val := str[strings.Index(str, "=")+1:]
+	val = val[1 : len(val)-1]
+	return val
+}
+
+func parseValSpace(str string) string {
+	val := str[strings.Index(str, " ")+1:]
+	return val
+}
+
+// OverwriteDefaultConfigFlags returns ConfigFlags with default values set when no
+// arg is passed, and the set arg for any passed values. The intent of this is to
+// overwrite the default flags with new defaults from the kuberc config file
+func (f *ConfigFlags) OverwriteDefaultConfigFlags(flags []string, args []string) {
```

Comment:
What I'd prefer if this is the only way to implement kuberc, don't support global config defaulting if this ^ is the only way :)

Reply:
Yeah I think this PR has highlighted some short comings with how kubectl handles global root level flags as a whole. I think I need to work on fixing that in a separate PR, similar to fixing the config subcommands. I need to investigate that further because, yes, writing this custom parser was janky and not fun.